src/libraries/System.Private.CoreLib/src/System/Numerics/Quaternion.cs

```diff
@@ -291,33 +271,59 @@ public static Quaternion Add(Quaternion value1, Quaternion value2)
         /// <returns>A new quaternion representing the concatenation of the <paramref name="value1" /> rotation followed by the <paramref name="value2" /> rotation.</returns>
         public static Quaternion Concatenate(Quaternion value1, Quaternion value2)
```

Comment:
This entire method should probably just be `=>value2 * value1` for simplicity and so we only need to optimize one code path.

Reply:
Makes sense